packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -139,6 +132,18 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
   };
 
   override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+
+    if(call.method == "addPlugin"){
+      try {
+        Amplify.addPlugin(AWSCognitoAuthPlugin())
+        LOG.info("Added Auth plugin")
+      } catch (e: Exception) {
+        handleAddPluginException("Auth", e, result)
+      }
+      result.success(null)
+      return
+    }
```

Comment:
Is there a reason to not handle this in the when statement below?

Reply:
If it's in the when statement it will trigger an error from the try catch around checkData @haverchuck

Reply:
ok